src/headphones2/postprocess/process_folder.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import unicode_literals
+
+import os
+
+import logbook
+
+from pathlib import Path
+from beets.library import Item
+from headphones2.postprocess.component_base import PostProcessorException
+from headphones2.postprocess.extensions.renamer import Renamer
+from headphones2.postprocess.taggers.acoustid_tagger import AcoustIDAlbumTagger
+from headphones2.postprocess.taggers.simple_tagger import SimpleBeetsTagger
+
+logger = logbook.Logger()
+
+
+def lookahead(iterable):
+    it = iter(iterable)
+    last = it.next()
+    for val in it:
+        yield last, False
+        last = val
+    yield last, True
+
+class FolderIterator(object):
```

Comment:
so - glob gave me lots of trouble due to its interal calling to fnmatch and re (very strage behavior) with directories that have [] chars in them for exmaple.

went with a simple filter approach:
    for ext in extensions:
        matches = p.listdir(fil=lambda x: x.ext == ext)
        file_pathes_to_process.extend(map(str, matches))

removed the folder class